lxd/main_callhook.go

```diff
@@ -24,14 +32,175 @@ func (c *cmdCallhook) Command() *cobra.Command {
+		err = os.MkdirAll(filepath.Dir(filepath.Join(containerRootFSMount, symlink.Link)), 0755)
+		if err != nil {
+			return fmt.Errorf("Failed to create the directory for the CDI symlink: %w\n", err)
+		}
+
+		// Create the symlink
+		err = os.Symlink(absTarget, filepath.Join(containerRootFSMount, symlink.Link))
+		if err != nil {
+			if !os.IsExist(err) {
+				return fmt.Errorf("Failed to create the CDI symlink: %w\n", err)
+			}
+
+			fmt.Printf("Symlink not created because link %q already exists for target %q\n", symlink.Link, absTarget)
+		}
+	}
+
+	// Updating the linker cache
+	l := len(hooks.LDCacheUpdates)
+	if l > 0 {
+		ldConfFilePath := fmt.Sprintf("%s/etc/ld.so.conf.d/%s", containerRootFSMount, customCDILinkerConfFile)
+		_, err = os.Stat(ldConfFilePath)
+		if err == nil {
+			// The file already exists. Read it first, analyze its entries
+			// and add the ones that are not already there.
+			ldConfFile, err := os.OpenFile(ldConfFilePath, os.O_APPEND|os.O_RDWR, 0644)
+			if err != nil {
+				return fmt.Errorf("Failed to open the ld.so.conf file at %s: %w\n", ldConfFilePath, err)
+			}
+
+			existingLinkerEntries := make(map[string]bool)
+			scanner := bufio.NewScanner(ldConfFile)
+			for scanner.Scan() {
+				existingLinkerEntries[strings.TrimSpace(scanner.Text())] = true
+			}
+
+			fmt.Printf("Existing linker entries: %v\n", existingLinkerEntries)
+			for _, update := range hooks.LDCacheUpdates {
+				if !existingLinkerEntries[update] {
+					fmt.Printf("Adding linker entry: %s\n", update)
+					_, err = fmt.Fprintln(ldConfFile, update)
+					if err != nil {
+						ldConfFile.Close()
+						return fmt.Errorf("Failed to write to the linker conf file at %s: %w\n", ldConfFilePath, err)
+					}
+
+					existingLinkerEntries[update] = true
+				}
+			}
+
+			ldConfFile.Close()
+		} else if errors.Is(err, os.ErrNotExist) {
+			// The file does not exist. We simply create it with our entries.
+			ldConfFile, err := os.OpenFile(ldConfFilePath, os.O_CREATE|os.O_WRONLY, 0644)
+			if err != nil {
+				return fmt.Errorf("Failed to create the linker conf file at %q: %w\n", ldConfFilePath, err)
+			}
+
+			for _, update := range hooks.LDCacheUpdates {
+				fmt.Printf("Adding linker entry: %s\n", update)
+				_, err = fmt.Fprintln(ldConfFile, update)
+				if err != nil {
+					ldConfFile.Close()
+					return fmt.Errorf("Failed to write to the linker conf file at %q: %w\n", ldConfFilePath, err)
+				}
+			}
+
+			ldConfFile.Close()
+		} else {
+			return fmt.Errorf("Could not stat the linker conf file to add CDI linker entries at %q: %w\n", ldConfFilePath, err)
+		}
+	}
+
+	// Then remove the linker cache and regenerate it
+	linkerCachePath := filepath.Join(containerRootFSMount, "/etc/ld.so.cache")
+	err = os.Remove(linkerCachePath)
+	if err != nil {
+		if !os.IsNotExist(err) {
+			return fmt.Errorf("Failed to remove the ld.so.cache file: %w\n", err)
+		}
+
+		fmt.Printf("%s file not removed because it doesn't exist\n", linkerCachePath)
```

Comment:
%q when putting strings into errors please


Also please can you add a prefix to the error so it doesn't start with the file path, e.g.

"Linker cache not found in %q, skipping removal"